ios/RNZendeskChatModule.m

```diff
@@ -201,4 +213,84 @@ - (void) dismissChatUI {
 	});
 }
 
+RCT_EXPORT_METHOD(sendMessage: (NSString*)message
+		resolver:(RCTPromiseResolveBlock)resolve
+		rejecter:(RCTPromiseRejectBlock)reject
+	) {
+		dispatch_sync(dispatch_get_main_queue(), ^{
+			[ZDKChat.chatProvider sendMessage:message completion:^(NSString *messageId, NSError *error) {
+				if (error) {
+					reject(@"error", @"no message id returned", error);
+				} else {
+					resolve(messageId);
+				}
+			}];
+		});
+}
+
+RCT_EXPORT_METHOD(sendChatRating: (NSString*)rating
+		resolver:(RCTPromiseResolveBlock)resolve
+		rejecter:(RCTPromiseRejectBlock)reject
+	) {
+        ZDKRating convertedRating = [RCTConvert ZDKRating:rating];
+
+		dispatch_sync(dispatch_get_main_queue(), ^{
+			[ZDKChat.chatProvider sendChatRating:convertedRating completion:^(ZDKRating _rating, NSError *error) {
+				if (error) {
+					reject(@"error", @"no rating returned", error);
+				} else {
+					resolve(rating);
+				}
+			}];
+		});
+}
+
+RCT_EXPORT_METHOD(endChat: (RCTPromiseResolveBlock)resolve
+		rejecter:(RCTPromiseRejectBlock)reject
+	) {
+		dispatch_sync(dispatch_get_main_queue(), ^{
+			[ZDKChat.chatProvider endChat:^(BOOL didEndChat, NSError *error) {
+				if (didEndChat) {
+					resolve(didEndChat ? @"true" : @"false");
```

Comment:
it was giving me an error returning `didEndChat` in the promise, so I returned strings instead... don't judge me

Reply:
```suggestion
					resolve(@(didEndChat));
```
Alternatively:
```objc
					resolve(didEndChat ? @YES : @NO);
```
For ancient historical reasons, `@YES` means `BOOL true, but wrapped in a Allocated Reference Box`
`@(…)` is "box the simple literal type into a standard `NSNumber` box."

Reply:
(and I believe these types are automatically understood by react-native's bridge serializer/deserializer)